kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/server/AdbServer.kt

```diff
@@ -24,8 +24,29 @@ interface AdbServer {
      * @throws AdbServerException if a result status of any command in @param commands is Failed
      * @return list of answers of commands' execution
      */
+    @Deprecated("This method doesn't work for the commands with the complex arguments containing " +
+            "whitespaces (e.g. 'adb pull \"path/with whitespace/file\") and doesn't allow commands piping like" +
+            "adbServer.performCmd(\"bash\", listOf(\"-c\", \"adb shell dumpsys deviceidle | grep mForceIdle\"))" +
+            "which the AdbServer.performCmd(String, List<String>) does. " +
+            "For more details, please check out https://github.com/KasperskyLab/Kaspresso/blob/master/docs/Wiki/Executing_adb_commands.en.md",
+        ReplaceWith("adbServer.performCmd(*commands, emptyList())")
+    )
     fun performCmd(vararg commands: String): List<String>
 
+    /**
+     * Performs shell commands blocking current thread. Allows more control over how arguments are parsed.
+     * Each list element is used as is. Refer to the https://docs.oracle.com/javase/8/docs/api/java/lang/ProcessBuilder.html.
```

Comment:
Please unify your comments. Somewhere you write class and method name, somewhere you provide a link.